app/helpers/application_helper.rb

```diff
@@ -1,2 +1,10 @@
 module ApplicationHelper
+  FULL_DATE_FORMAT_STRING= '%d %B %Y'
+  MONTH_YEAR_DATE_FORMAT_STRING= '%m/%Y'
+
+  ['full', 'month_year'].each do |type_of|
+    define_method "format_#{type_of}_date" do |date|
+      date.try :strftime, %Q~#{ApplicationHelper.const_get("#{type_of.upcase}_DATE_FORMAT_STRING".to_sym)}~
```

Comment:
`.try` is considered a code smell as it effectively says "I have no idea what's going on here but I'll give it a shot". Here, specifically, I think it also can be harmful by letting callers of this method get away with too much: The method is clearly about time conversion. There may be times when it's called with `nil` if a particular date attribute is missing. Using a true Null Object helps to avoid type checking in the views in such cases, which is great - we don't have to write
```erb
<% if @sponsorship.end_date %>
  <%= format_month_year_date @sponsorship.end_date %>
<% end %>
```
However, allowing the method to be called with _any_ object type without blowing up can lead to sloppiness and difficult-to-track bugs. With that in mind, I still prefer the Null Object implementation we'd come up with in the comments earlier.